pint-abi/src/encode.rs

```diff
@@ -18,7 +18,22 @@ impl Encode for bool {
     }
 }
 
-impl<T: Encode> Encode for [T] {
+impl<T: Encode + Default> Encode for Option<T> {
+    fn encode<W: Write>(&self, w: &mut W) -> Result<(), W::Error> {
+        match self {
+            Some(val) => {
+                val.encode(w)?; // data
+                w.write_word(1) // tag
+            }
+            None => {
+                T::default().encode(w)?; // data
+                w.write_word(0) // tag
+            }
+        }
+    }
+}
+
+impl<T: Encode + Default> Encode for [T] {
```

Comment:
I don't think there's any need to add the `Default` bound to the other `Encode` implementations - I think it's only relevant to `Option<T>`'s `Encode` implementation.

Reply:
Yeah good point! Fixed :)